collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -17,56 +17,63 @@
 import android.annotation.SuppressLint;
 import android.content.Context;
 import android.net.Uri;
+import android.util.TypedValue;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
 import android.widget.Button;
-import android.widget.LinearLayout;
 import android.widget.TextView;
 import android.widget.Toast;
 
 import org.javarosa.core.model.data.IAnswerData;
 import org.javarosa.core.model.data.StringData;
+import org.javarosa.form.api.FormEntryPrompt;
 import org.odk.collect.android.R;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.formentry.questions.WidgetViewUtils;
 import org.odk.collect.android.utilities.CustomTabHelper;
-import org.odk.collect.android.widgets.interfaces.ButtonWidget;
-
-import static org.odk.collect.android.formentry.questions.WidgetViewUtils.createSimpleButton;
-import static org.odk.collect.android.formentry.questions.WidgetViewUtils.getCenteredAnswerTextView;
+import org.odk.collect.android.utilities.MultiClickGuard;
 
 /**
  * Widget that allows user to open URLs from within the form
  *
  * @author Yaw Anokwa (yanokwa@gmail.com)
  */
 @SuppressLint("ViewConstructor")
-public class UrlWidget extends QuestionWidget implements ButtonWidget {
+public class UrlWidget extends QuestionWidget {
 
-    private Uri uri;
-    final Button openUrlButton;
-    final TextView stringAnswer;
     private final CustomTabHelper customTabHelper;
+    private Uri uri;
```

Comment:
I would get rid of this field. this is used in just one place `onButtonClick()` so it would be better to add a method like:
```
    private Uri getUri() {
        return Uri.parse(stringAnswer.getText().toString());
    }
```